tiledb/sm/subarray/subarray.h

```diff
@@ -245,10 +246,29 @@ class Subarray {
   void compute_range_offsets();
 
   /**
-   * Computes the tile overlap with all subarray ranges for
-   * all fragments.
-   */
-  Status compute_tile_overlap(ThreadPool* compute_tp);
+   * Precomputes the tile overlap with all subarray ranges for
+   * all fragments. The state is cached internally and accessible
+   * through the `get_subarray_tile_overlap` API.
+   *
+   * This routine may not compute tile overlap for the entire range.
+   * This only guarantees that:
+   *   1. Tile overlap will be computed for at least one range.
+   *   2. Tile overlap is computed starting from `start_range_idx`.
+   *
+   * The caller is responsible for checking the range indexes that
+   * were computed through the `get_subarray_tile_overlap` API.
+   *
+   * @param start_range_idx The start range index.
+   * @param start_range_idx The target end range index.
+   * @param compute_tp The compute thread pool.
+   * @param override_memory_constraint When true, this forces the
+   *    routine to compute tile overlap for all ranges.
+   */
+  Status precompute_tile_overlap(
```

Comment:
For the C API we may want to throw an error and document it appropriately in `tiledb.h`. Just an idea, I am not absolute about that.

Reply:
We will ticket work to refactor the C API path so that it does not exceed the memory budget. This patch leaves that path unchanged, so I suggest we leave this as-is and follow up the real fix.